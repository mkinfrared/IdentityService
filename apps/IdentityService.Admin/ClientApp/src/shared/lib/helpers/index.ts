export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { FetchClient, FetchError, isFetchError } from "./fetchClient";

export { lazyImport } from "./lazyImport";

export { userManager } from "./userManager";
