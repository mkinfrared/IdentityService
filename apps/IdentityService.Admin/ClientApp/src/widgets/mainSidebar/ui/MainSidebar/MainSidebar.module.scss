.MainSidebar {
  background-color: cornflowerblue;
  flex: 1 0 280px;
  max-width: 280px;

  a {
    display: block;
    text-decoration: none;
  }

  .listItem {
    display: block;
    width: 100%;
  }
}
