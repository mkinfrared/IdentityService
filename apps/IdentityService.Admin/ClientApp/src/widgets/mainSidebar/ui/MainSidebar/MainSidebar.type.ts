import { ReactNode } from "react";

export type MainSidebarProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};

export type LinkItem = {
  text: string;
  path: string;
  icon: ReactNode;
};
