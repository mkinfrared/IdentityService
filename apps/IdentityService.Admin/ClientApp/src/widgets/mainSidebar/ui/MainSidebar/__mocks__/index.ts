export { default } from "./MainSidebar.mock";
