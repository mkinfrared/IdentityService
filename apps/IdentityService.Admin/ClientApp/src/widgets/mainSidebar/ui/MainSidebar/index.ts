export { default } from "./MainSidebar";

export * from "./MainSidebar.type";
