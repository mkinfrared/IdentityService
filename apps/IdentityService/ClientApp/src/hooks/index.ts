export * from "./useAppNavigate";

export * from "./useConsentQuery";

export * from "./useReturnUrl";
