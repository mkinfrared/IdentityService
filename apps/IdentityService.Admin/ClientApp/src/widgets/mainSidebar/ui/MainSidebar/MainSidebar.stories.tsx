import { Meta, Story } from "@storybook/react";

import MainSidebar from "./MainSidebar";
import { MainSidebarProps } from "./MainSidebar.type";
import css from "./Story.module.scss";

export default {
  title: "COMPONENTS/MainSidebar",
  component: MainSidebar,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<MainSidebarProps> = (args) => (
  <div className={css.Story}>
    <MainSidebar {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
