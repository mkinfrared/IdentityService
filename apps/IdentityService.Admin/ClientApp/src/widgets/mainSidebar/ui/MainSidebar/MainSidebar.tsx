/* eslint-disable react/no-array-index-key */

import { classNames } from "@identity-service/core";
import { LanguageSharp } from "@mui/icons-material";
import {
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
} from "@mui/material";
import { useNavigate } from "react-router-dom";

import { useUserSession } from "shared/lib/hooks";
import { APP_ROUTES } from "shared/types/routes";

import css from "./MainSidebar.module.scss";
import { LinkItem, MainSidebarProps } from "./MainSidebar.type";

const MainSidebar = ({ className }: MainSidebarProps) => {
  const user = useUserSession();
  const navigate = useNavigate();

  const linkItems: LinkItem[] = [
    {
      text: "Clients",
      path: APP_ROUTES.Clients.relativePath,
      icon: <LanguageSharp />,
    },
    {
      text: "Identity Resources",
      path: APP_ROUTES.IdentityResources.relativePath,
      icon: <LanguageSharp />,
    },
    {
      text: "Api Resources",
      path: APP_ROUTES.ApiResources.relativePath,
      icon: <LanguageSharp />,
    },
    {
      text: "Api Scopes",
      path: APP_ROUTES.ApiScopes.relativePath,
      icon: <LanguageSharp />,
    },
    {
      text: "Persisted Grants",
      path: APP_ROUTES.PersistedGrants.relativePath,
      icon: <LanguageSharp />,
    },
    {
      text: "Users",
      path: APP_ROUTES.Users.relativePath,
      icon: <LanguageSharp />,
    },
    {
      text: "Roles",
      path: APP_ROUTES.Roles.relativePath,
      icon: <LanguageSharp />,
    },
  ];

  const handleItemClick = (path: string) => () => {
    navigate(path, { replace: false });
  };

  if (!user) {
    return null;
  }

  return (
    <aside
      className={classNames(css.MainSidebar, className)}
      data-testid="MainSidebar"
    >
      <div>
        <List>
          {linkItems.map(({ text, path, icon }, index) => (
            <ListItem
              key={index}
              disablePadding
              onClick={handleItemClick(path)}
            >
              <ListItemButton>
                <ListItemIcon>{icon}</ListItemIcon>
                <ListItemText primary={text} />
              </ListItemButton>
            </ListItem>
          ))}
        </List>
      </div>
    </aside>
  );
};

export default MainSidebar;
