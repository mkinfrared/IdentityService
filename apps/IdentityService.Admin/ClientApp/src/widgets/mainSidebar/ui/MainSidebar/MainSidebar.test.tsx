import { render } from "@testing-library/react";

import { useUserSession } from "shared/lib/hooks";

import MainSidebar from "./MainSidebar";

jest.mock("shared/lib/hooks");

describe("<MainSidebar />", () => {
  const useUserSessionMock = jest.mocked(useUserSession);

  beforeEach(() => {
    useUserSessionMock.mockReturnValue({ bar: "foo" } as any);
  });

  const Component = <MainSidebar />;

  it("should be defined", () => {
    expect(MainSidebar).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("MainSidebar");

    expect(element).toBeDefined();
  });
});
