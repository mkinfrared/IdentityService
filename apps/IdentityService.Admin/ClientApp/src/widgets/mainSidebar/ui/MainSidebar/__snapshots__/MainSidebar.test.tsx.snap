// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSidebar /> should match the snapshot 1`] = `
<div>
  <aside
    class="MainSidebar"
    data-testid="MainSidebar"
  >
    <li>
      Clients
       
      clients
    </li>
  </aside>
</div>
`;
